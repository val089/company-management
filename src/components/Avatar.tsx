import { StyleSheet, View, Image } from 'react-native';
import avatarImg from '@app/assets/images/avatar.png';

type AvatarProps = {
  imageUri?: string;
};

export const Avatar = ({ imageUri }: AvatarProps) => {
  return (
    <>
      {imageUri ? (
        <View style={styles.imgContainer}>
          <Image source={{ uri: imageUri }} style={styles.image} />
        </View>
      ) : (
        <Image source={avatarImg} style={styles.image} />
      )}
    </>
  );
};

const styles = StyleSheet.create({
  imgContainer: {
    overflow: 'hidden',
    borderRadius: 50,
  },
  image: {
    width: 80,
    height: 84,
  },
});
