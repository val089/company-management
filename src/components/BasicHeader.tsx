import { Pressable, StyleSheet, View, ViewStyle } from 'react-native';
import { TextStyle } from 'react-native';
import { SafeAreaProvider, SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context';
import { AddIcon } from '@app/assets/icons/AddIcon';
import { BackArrowIcon } from '@app/assets/icons/BackArrowIcon';
import { bgColor, fontColor, GlobalStyles } from '@app/constants/styles';
import { useNavigation } from '@react-navigation/native';

import { Typography } from './Typography';

type BasicHeaderProps = {
  title?: string;
  isBackIcon?: boolean;
  onPlusPress?: () => void;
};

export const BasicHeader = ({ title = '', isBackIcon = true, onPlusPress }: BasicHeaderProps) => {
  const navigation = useNavigation();
  const insets = useSafeAreaInsets();

  const headerStyle: ViewStyle = {
    justifyContent: isBackIcon ? 'flex-start' : 'space-between',
  };

  const labelStyle: TextStyle = {
    paddingLeft: isBackIcon ? 20 : 0,
  };

  return (
    <SafeAreaProvider>
      <SafeAreaView>
        <View style={[styles.header, headerStyle]}>
          {isBackIcon && (
            <Pressable onPress={() => navigation.goBack()} hitSlop={10}>
              <BackArrowIcon fill={fontColor} />
            </Pressable>
          )}
          <Typography type="large" style={[styles.title, labelStyle]}>
            {title}
          </Typography>
          {onPlusPress && (
            <Pressable onPress={onPlusPress}>
              <AddIcon />
            </Pressable>
          )}
        </View>
      </SafeAreaView>
    </SafeAreaProvider>
  );
};

const styles = StyleSheet.create({
  header: {
    height: 70,
    paddingHorizontal: 16,
    backgroundColor: bgColor,
    alignItems: 'center',
    flexDirection: 'row',
    elevation: 20,
    shadowColor: GlobalStyles.colors.black,
    shadowOffset: { width: -2, height: 5 },
    shadowOpacity: 0.1,
    shadowRadius: 3,
  },
  title: {
    fontWeight: 'bold',
    color: GlobalStyles.colors.blue100,
  },
});
