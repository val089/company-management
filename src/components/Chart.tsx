import { Dimensions, StyleSheet, View } from 'react-native';
import { BarChart } from 'react-native-chart-kit';
import { GlobalStyles } from '@app/constants/styles';
import { fontColor } from '@app/constants/styles';
import { Expense } from '@app/types';
import { formatDataForChart } from '@app/utils/formatDataForChart';
import { getTimestampFirstDayOfMonthsAgo } from '@app/utils/getTimestampFirstDayOfMonthsAgo';

import { Typography } from './Typography';

interface ChartProps {
  expenses: Expense[];
}

const filterAndFormatExpenses = (expenses: Expense[]) => {
  const firstDayOfMonthsAgo = getTimestampFirstDayOfMonthsAgo(2);
  const filteredExpenses = expenses
    .filter(({ createdAt }) => createdAt >= firstDayOfMonthsAgo)
    .map(({ createdAt, amount, type }) => {
      const date = new Date(createdAt);
      return {
        month: date.toLocaleString('en', { month: 'long' }),
        amount: type === 'expense' ? -amount : amount,
      };
    });

  return formatDataForChart(filteredExpenses);
};

export const Chart = ({ expenses = [] }: ChartProps) => {
  if (!expenses || expenses.length === 0) {
    return (
      <Typography type="normal" style={styles.info}>
        No expenses to display chart
      </Typography>
    );
  }

  const formatedDataForChart = filterAndFormatExpenses(expenses);

  const chartLabels = formatedDataForChart.map(({ month }) => month);
  const chartData = formatedDataForChart.map(({ amount }) => amount);

  const data = {
    labels: chartLabels,
    datasets: [
      {
        data: chartData,
      },
    ],
  };

  return (
    <>
      <Typography type="normal" style={styles.sectionTitle}>
        Revenue during thwo months
      </Typography>

      <View style={styles.container}>
        <BarChart
          segments={4}
          fromZero
          style={styles.chart}
          data={data}
          width={Dimensions.get('window').width}
          height={300}
          chartConfig={{
            decimalPlaces: 0,
            backgroundGradientFrom: GlobalStyles.colors.blue100,
            backgroundGradientFromOpacity: 0.95,
            backgroundGradientTo: GlobalStyles.colors.blue100,
            backgroundGradientToOpacity: 1,
            color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
            labelColor: (opacity = 10) => `rgba(255, 255, 255, ${opacity})`,
          }}
          verticalLabelRotation={0}
          yAxisLabel="$"
          yAxisSuffix=""
        />
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  chart: {
    paddingTop: 20,
  },
  filters: {
    flexDirection: 'row',
    marginBottom: 10,
  },
  filterBtn: {
    backgroundColor: GlobalStyles.colors.blue100,
    borderRadius: 10,
    paddingVertical: 5,
    paddingHorizontal: 10,
    marginHorizontal: 5,
  },
  filterText: {
    color: GlobalStyles.colors.white,
  },
  sectionTitle: {
    paddingTop: 16,
    fontWeight: 'bold',
    paddingHorizontal: 16,
    color: fontColor,
  },
  info: {
    paddingHorizontal: 16,
    paddingTop: 32,
  },
});
