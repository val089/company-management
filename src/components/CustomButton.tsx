import { Pressable, StyleProp, StyleSheet, ViewStyle, PressableProps, View } from 'react-native';
import { GlobalStyles } from '@app/constants/styles';

interface CustomButtonProps extends PressableProps {
  style?: StyleProp<ViewStyle>;
  onPress: () => void;
}

export const CustomButton = ({
  children,
  onPress,
  style,
  disabled,
  ...restProps
}: CustomButtonProps) => {
  return (
    <View style={[styles.buttonContainer, style]}>
      <Pressable
        style={[
          styles.button,
          { backgroundColor: disabled ? GlobalStyles.colors.grey700 : GlobalStyles.colors.blue100 },
        ]}
        onPress={onPress}
        android_ripple={{ color: GlobalStyles.colors.blue100 }}
        {...restProps}>
        {children}
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    overflow: 'hidden',
    borderRadius: 50,
  },
  button: {
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    color: '#fff',
  },
});
