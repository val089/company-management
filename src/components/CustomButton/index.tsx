import { Pressable, StyleProp, StyleSheet, ViewStyle, PressableProps, View } from 'react-native';
import { GlobalStyles } from '../../constants/styles';

interface CustomButtonProps extends PressableProps {
  style?: StyleProp<ViewStyle>;
  onPress: () => void;
}

export const CustomButton = ({ children, onPress, style, ...restProps }: CustomButtonProps) => {
  return (
    <View style={[styles.buttonContainer, style]}>
      <Pressable
        style={styles.button}
        onPress={onPress}
        {...restProps}
        android_ripple={{ color: GlobalStyles.colors.blue100 }}>
        {children}
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    overflow: 'hidden',
    borderRadius: 50,
  },
  button: {
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    color: '#fff',
    backgroundColor: GlobalStyles.colors.secondary100,
  },
});
