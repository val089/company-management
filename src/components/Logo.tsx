import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native';
import { Typography } from './Typography';

interface LogoProps {
  style?: StyleProp<ViewStyle>;
  fontSize?: number;
}

export const Logo = ({ style, fontSize = 40 }: LogoProps) => {
  return (
    <View style={[styles.logoContainer, style]}>
      <Typography type="logoText" style={[styles.text, { fontSize }]}>
        Company
      </Typography>
      <Typography type="logoText" style={[styles.text, { fontSize }]}>
        Management
      </Typography>
    </View>
  );
};

const styles = StyleSheet.create({
  logoContainer: {},
  text: {
    lineHeight: 45,
  },
});
