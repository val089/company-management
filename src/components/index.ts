export * from './AddImageModal';
export * from './Avatar';
export * from './BasicHeader';
export * from './Chart';
export * from './CustomButton';
export * from './HomeHeader';
export * from './Input';
export * from './LanguageDropdown';
export * from './LoadingOverlay';
export * from './Logo';
export * from './ModalPopup';
export * from './MoneySummary';
export * from './Radio';
export * from './Select';
export * from './TextFieldButton';
export * from './Typography';
