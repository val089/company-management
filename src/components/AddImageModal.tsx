import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

import { ModalPopup } from '@app/components/ModalPopup';
import { CustomButton } from './CustomButton';
import { Typography } from './Typography';

type ConfirmDeletionModalProps = {
  isModalOpen: boolean;
  hideModal: () => void;
  choosePhoto: () => void;
  openCamera: () => void;
};

export const AddImageModal = ({
  isModalOpen,
  hideModal,
  choosePhoto,
  openCamera,
}: ConfirmDeletionModalProps) => {
  return (
    <ModalPopup
      visible={isModalOpen}
      onCancel={hideModal}
      header={<Typography type="normal">Add a photo</Typography>}
      footer={null}>
      <View style={styles.content}>
        <CustomButton onPress={choosePhoto}>
          <Text>Open Gallery</Text>
        </CustomButton>

        <CustomButton onPress={openCamera} style={styles.openCameraBtn}>
          <Text>Open Camera</Text>
        </CustomButton>
      </View>
    </ModalPopup>
  );
};

const styles = StyleSheet.create({
  iconContainer: {
    alignItems: 'center',
  },
  content: {
    marginVertical: 40,
  },
  openCameraBtn: {
    marginTop: 20,
  },
});
