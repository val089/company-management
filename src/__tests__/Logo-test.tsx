import renderer from 'react-test-renderer';
import { Logo } from '@app/components/Logo';

import 'react-native';

test('renders correctly the Logo component', () => {
  const tree = renderer.create(<Logo />).toJSON();
  expect(tree).toMatchSnapshot();
});
