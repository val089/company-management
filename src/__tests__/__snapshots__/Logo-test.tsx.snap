// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the Logo component 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Poppins-Bold",
          "fontSize": 40,
        },
        Array [
          Object {
            "lineHeight": 45,
          },
          Object {
            "fontSize": 40,
          },
        ],
      ]
    }
  >
    Company
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Poppins-Bold",
          "fontSize": 40,
        },
        Array [
          Object {
            "lineHeight": 45,
          },
          Object {
            "fontSize": 40,
          },
        ],
      ]
    }
  >
    Management
  </Text>
</View>
`;
